device.go

```diff
@@ -158,11 +166,55 @@ func newBlockDevice(devInfo DeviceInfo) *BlockDevice {
 	}
 }
 
-func (device *BlockDevice) attach(h hypervisor) error {
+func (device *BlockDevice) attach(h hypervisor, c *Container, create bool) error {
+	device.DeviceInfo.ID = uuid.Generate().String()
+	devID := fmt.Sprintf("drive-%s", device.DeviceInfo.ID)
+
+	drive := Drive{
+		File:   device.DeviceInfo.HostPath,
+		Format: "raw",
+		ID:     devID,
+	}
+
+	// Increment the block index for the pod. This is used to determine the name
+	// for the block device in the case where the block device is used as container
+	// rootfs and the predicted block device name needs to be provided to the agent.
+	_, err := c.pod.getAndSetPodBlockIndex()
+	if err != nil {
+		return err
+	}
+
+	virtLog.Infof("Attaching block device %s", device.DeviceInfo.HostPath)
+	if create {
+		if err := h.addDevice(drive, blockDev); err != nil {
+			return err
+		}
+		device.DeviceInfo.Hotplugged = false
+	} else {
+		if err := h.hotplugAddDevice(drive, blockDev); err != nil {
```

Comment:
be aware of bus option, if you  do not specify a bus qemu will hot plug the device directly in the PCI, for pc that's ok because it's ACPI based, but for q35 this will fail, you have to specify a pcie root port or a downstream port.
This will become a problem since we want to support both q35 and pc

Reply:
Thanks for noticing this @devimc !
That's important that we stay compatible with both so that we can switch back if needed, and also for performance comparison. Anyway, I think we all agree we need something more modular to support both machine types.

Reply:
@devimc Lets add support for q35 in a separate PR.

Reply:
@amshinde @sboeuf  yep, working on that :)

Reply:
@devimc @amshinde let us first add support in the current ACPI implementation. Let the q35 land seperately.